### PR TITLE
Buttons

### PR DESCRIPTION
**Pull request by romangorelik**

### CODE REVIEW

File: src/content/buttons/index.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            </Button>
+            </div>
+            {code && <Highlight language="jsx" className={classes.code}>
+            {buttonString}
+            </Highlight>}
+        </section>
+    )
+}
+
+export default withStyles(styles)(Buttons)
```
Comment (herrethan) [inline src/content/buttons/index.tsx]: nit: extra spaces

File: content/components/buttons.mdx [hunk 2]
```diff
@@ -2,10 +2,18 @@
...
 At imperdiet dui accumsan sit. Arcu dui vivamus arcu felis bibendum ut tristique et. Pharetra et ultrices neque ornare aenean euismod. Dignissim enim sit amet venenatis urna. Lacus luctus accumsan tortor posuere. Aliquet porttitor lacus luctus accumsan. Sit amet consectetur adipiscing elit. Pellentesque eu tincidunt tortor aliquam nulla facilisi cras fermentum. Cras semper auctor neque vitae tempus quam. Lobortis mattis aliquam faucibus purus in massa. Morbi tristique senectus et netus. Quam adipiscing vitae proin sagittis nisl. Quis blandit turpis cursus in hac habitasse platea dictumst quisque. Mauris cursus mattis molestie a iaculis at erat pellentesque.
 
-```
-i am a code snippet
-```
+### Examples 
+import Buttons from 'content/buttons/index.tsx'
+
+<Buttons/>
+
```
Comment (herrethan) [inline content/components/buttons.mdx]: I think we should make more of this customizable by editors, see below

File: src/content/buttons/index.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+             color="primary"
+             startIcon={<DeleteIcon />}
+             className={classes.margin}
+            >
+                Icon and Text
+            </Button>
+            </div>
+            {code && <Highlight language="jsx" className={classes.code}>
+            {buttonString}
+            </Highlight>}
```
Comment (herrethan) [inline src/content/buttons/index.tsx]: ☝️ this is the code that I think should be in the actual .md file.
something like
```jsx
<CodeSample>
  <Button color="primary" variant="outlined" size="medium"> Outlined </Button>
  ...
</CodeSample>
```
Where the `CodeSample` component would handle the styling, toggle buttton, and string converting all internally. Let me know if that makes sense.

That way editors can put whatever buttons they want in there, and the strings are guaranteed to be an exact representation.

File: src/content/buttons/string.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+<Button size="medium" variant="outlined" color="primary"> Light </Button>
+
+<Button size="medium" variant="outlined" color="primary"> Outlined </Button>
+
+<Button size="medium" color="primary"> Text </Button>
+
+<Button color="primary" startIcon={<DeleteIcon />}> Icon and Text </Button>
+`
+
+export default buttonString
```
Comment (herrethan) [inline src/content/buttons/string.ts]: Have you looked into libraries that can stringify jsx?
This looks like a good one https://www.npmjs.com/package/jsx-to-string.

File: package.json [hunk 5]
```diff
@@ -83,6 +83,7 @@
     "gatsby-mdx": "^0.3.4",
     "gatsby-plugin-catch-links": "^2.0.9",
     "gatsby-plugin-manifest": "^2.0.9",
+    "gatsby-plugin-mdx": "^1.0.67",
```
Comment (herrethan) [inline package.json]: ahh, is this the plugin that allows `import`s in .mdx files? i was looking for that.

also, did you make sure the cms still runs properly with these updates?

File: src/content/buttons/index.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  },
+  code: {
+    padding: `16px`,
+    fontSize: `14px`
+  }
+});
+
+const Buttons = (props: IButtons) => {
+  const classes = useStyles();
+  const [code, setCode] = useState(false)
```
Comment (jdents) [inline src/content/buttons/index.tsx]: Can you prefix this with `is`?

File: src/content/buttons/index.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  const children = React.Children.toArray(props.children)
+
+  let stringChildren: any = []
+
+  for (let i = 0; i < React.Children.count(children); i++) {
+    stringChildren.push(jsxToString(props.children[i]).replace('WithStyles(ForwardRef(Button))', 'Button').replace('/WithStyles(ForwardRef(Button))', '/Button'))
+  }
+
+  stringChildren = stringChildren.join("\n\n")
```
Comment (jdents) [inline src/content/buttons/index.tsx]: This should probably be memoized.
```jsx
const stringChildren = useMemo(() => {
  let stringed: any = []

  for (let i = 0; i < React.Children.count(children); i++) {
    stringed
      .push(jsxToString(props.children[i])
      .replace('WithStyles(ForwardRef(Button))', 'Button')
      .replace('/WithStyles(ForwardRef(Button))', '/Button'))
  }

  return stringed.join("\n\n")
}, [props.children])
```

Comment (romangorelik) [reply src/content/buttons/index.tsx]: good call, changed

**Comment by herrethan** [comment]
PR NuMbEr oNe! 🔥

**Pull request merged**